### PR TITLE
fix: empty n:m relationships loads every record instead of an empty a…

### LINKED ISSUE #258: Empty N:M Relationships Loads Every Record Instead of an Empty Array
When a many-to-many relationship has no associated records it appears that all of the associated `Model`'s records are loaded instead.

As strange as this bug is it appears to generate the correct query at first and then runs an additional query after an empty array is returned.

--

#### First Query

```sql
SELECT "tags"."id" AS "id", "categorizations"."tag_id" AS "tagId", "categorizations"."post_id" AS "postId" FROM "tags" INNER JOIN "categorizations" ON "tags"."id" = "categorizations"."tag_id" WHERE "categorizations"."post_id" IN (1)
```

#### Second Query

```sql
SELECT "tags"."id" AS "id", "tags"."name" AS "name", "tags"."created_at" AS "createdAt", "tags"."updated_at" AS "updatedAt" FROM "tags"
```

### PR DESCRIPTION
**Pull request by zacharygolba**
Closes #258 

--

screen shot 2016-08-06 at 10 59 14 am
screen shot 2016-08-06 at 10 59 03 am

### CODE REVIEW

File: src/packages/database/query/utils/build-results.js [hunk 1]
```diff
@@ -84,18 +84,17 @@ export default async function buildResults<T: Model>({
...
+            Reflect.set(record, name, relatedResults.filter(({
+              rawColumnData
+            }) => {
               const fk = Reflect.get(rawColumnData, foreignKey);
               const pk = Reflect.get(record, model.primaryKey);
 
               return fk === pk;
-            });
-
-            if (match.length) {
```
Comment (zacharygolba) [inline src/packages/database/query/utils/build-results.js]: Not setting the relationship to an empty array is what caused duplicate fetching for eager loaded relationships w/o any associated records.

File: src/packages/database/relationship/utils/getters.js [hunk 2]
```diff
@@ -1,28 +1,63 @@
 // @flow
+import { camelize } from 'inflection';
+
 import type { Model } from '../../index';
 import type { Relationship$opts } from '../index';
 
 /**
  * @private
  */
+async function getHasManyThrough(owner: Model, {
```
Comment (zacharygolba) [inline src/packages/database/relationship/utils/getters.js]: Not properly fetching `hasManyThrough` relationships is what caused all records to be returned as opposed to the actual value for the second (lazy loaded) relationship query that occurred due to the bug in the line comment above.

**Comment by mutewinter** [comment]
Why wasn't this caught by tests? Just an edge case?

**Comment by zacharygolba** [comment]
Sadly the test app doesn't contain any n:m relationships. I usually use the [social-network](https://github.com/postlight/lux/tree/master/examples/social-network) when I work on these features.

> I haven't been super strict about tests lately because I'm saving a lot of that work for #77.

What I really mean is, I've been getting a bit lazy and #77 should be a higher priority. 😉
